receiver/otlpreceiver/internal/trace/otlp.go

```diff
@@ -49,7 +51,30 @@ func (r *Receiver) Export(ctx context.Context, req ptraceotlp.ExportRequest) (pt
 	// NonPermanent errors will be converted to codes.Unavailable (equivalent to HTTP 503)
 	// Permanent errors will be converted to codes.InvalidArgument (equivalent to HTTP 400)
 	if err != nil {
-		return ptraceotlp.NewExportResponse(), errors.GetStatusFromError(err)
+		// See if an error is retryable.
+		cet := consumererror.Traces{}
+		if errors.As(err, &cet) {
+			retries := 3
+
+			// Retry three times; the real logic would be more complex akin to consumerretry.
+			for i := 0; i < retries; i++ {
```

Comment:
This is an extremely rudimentary reproduction of the consumerretry package, where I think this error parsing would ultimately live: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/coreinternal/consumerretry/logs.go#L61.